### PR TITLE
R2-2317 Document SSE-C Compatibility on R2

### PR DESCRIPTION
**Pull request by helloimalastair**
### Summary

Updated Compatibility Matrix, API Reference, and added Example.

Please merge when feature lands.

### Preview

- [https://developers.cloudflare.com/r2/api/s3/api/](https://sse-c.cfdocs-alastair.pages.dev/r2/api/s3/api/)
- [https://developers.cloudflare.com/r2/examples/ssec/](https://sse-c.cfdocs-alastair.pages.dev/r2/examples/ssec/)
- [https://developers.cloudflare.com/r2/api/workers/workers-api-reference/](https://sse-c.cfdocs-alastair.pages.dev/r2/api/workers/workers-api-reference/)

### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.

### CODE REVIEW

File: src/content/docs/r2/examples/ssec.mdx [hunk 1]
```diff
@@ -0,0 +1,286 @@
+---
+title: Using SSE-C
```
Comment (dcpena) [inline src/content/docs/r2/examples/ssec.mdx]:
```suggestion
title: Use SSE-C
```

File: src/content/docs/r2/examples/ssec.mdx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+difficulty: Intermediate
+content_type: 📝 Tutorial
+updated: 2024-09-27
+---
+
+import { Tabs, TabItem } from "~/components";
+
+The following tutorial shows some snippets for how to use Server-Side Encryption with Customer-Provided Keys(SSE-C) on R2.
+
+## Things to note
```
Comment (dcpena) [inline src/content/docs/r2/examples/ssec.mdx]:
```suggestion
## Before you begin
```

**Review by dcpena — APPROVED** [review]
Two small suggestions :) Also, thanks for fixing all of the API content!

**Review by dcpena — APPROVED** [review]

**Comment by dcpena** [comment]
This content was created before Mikkel's internship ended, but the code hasn't been merged just yet. Leaving open until I get PM go-ahead.

**Review by jonesphillip — APPROVED** [review]

**Comment by helloimalastair** [comment]
Please hold until runtime update is released

**Comment by dcpena** [comment]
@helloimalastair I added you to the group that gives you merge permissions. You should be able to merge your PR whenever it's ready. 😁

**Pull request merged**